### PR TITLE
Hour-of-Code route

### LINKED ISSUE #4705: [CHORE] Add tracking pixel
We need to add a tracking pixel from Hour of Code (https://code.org/api/hour/begin_hedy_text.png) somewhere on the link we will create for them, info:

**Tracking Pixel**
In order to help us more accurately track participation in the Hour of Code, please include a 1-pixel tracking image on the first page of your Hour of Code activity. Place the pixel-image on the start page (if you have a landing page before the real start of your activity, place the pixel-image on the start page instead of the landing page). Please do not place the pixel on any other page. 

If an activity is only offered as an app download, or a lesson plan download (or link to a PDF or Google Doc), there is no need to include the tracking pixel and you can ignore this 

I am still double checking the link, will update later, but we can start to think about how/where to add it.

### LINKED ISSUE #4706: [UI idea] Create link based customization for hour of code
We want to create something like this link: https://hedy.org/hour-of-code/1?keyword_language=da&language=da

Which then shows a subset of adventures. 

Where to define the set? We could do it in the code in hedy_content.py for now? An alternative could be to add tags to adventures in the yaml, and load based on those. That is more flexible and future proof, but might require more work.

I think we now also really need to pick up #4606 that @TiBiBa was working on since people coming to Hedy from the outside will surely be confused by that legacy button!

### PR DESCRIPTION
**Pull request by hasan-sh**
Fixes #4706 and fixes #4705

Adventures subsets can now be passed through the route. For instance, the following route will give back the corresponding translated adventures of level 8: 
> /hour-of-code/8?**subset=fortune,turtle**&language=en

Also, the `keyword_language` and `language` parameters work as expected. 

@SabinaChita, this part is relevant for the newsletter:

**This PR also changes the button from "level X" to "next adventure name"**

### CODE REVIEW

**Comment by Felienne** [comment]
> Adventures subsets can now be passed through the route. For instance, the following route will give back the corresponding translated adventures of level 8:
> 
> > /hour-of-code/8?**subset=fortune,turtle**&language=en

Ah cool, that will allow for a lot of other customization as well (f.e. for a workshop!)
 
> Also, the `keyword_language` and `language` paramenters work as expected.

Great!

**Comment by Felienne** [comment]
I think we can merge this for now, including the quiz, we can always remove that later-on.

**Review by Felienne — CHANGES_REQUESTED** [review]
Hi @hasan-sh!

I just noticed we show the puzzle, not the quiz in the route:

image

The puzzles aren't the most awesome part of the UI at this point, they have a few annoying issues, so I don't want to showcase those at the moment. Can we swap the puzzle for the quiz? With that this should be good to go!

File: app.py
Comment (Felienne) [inline app.py:1257]: I do think we want to have the button in this view too (once is has changed into "next adventure" where appropriate)

File: app.py
Comment (Felienne) [inline app.py:1261]: This way of disabling parameters by commenting them out is not what we commonly do (I think setting the variable to False occurs more often) I actually think what you do here is more clear, you have the information that it is not in use right in view, just want to point out that this is not what is used elsewhere (maybe it should be?)

File: app.py
Comment (hasan-sh) [reply app.py:1257]: I will merge #4709 once it's done.

Comment (hasan-sh) [reply app.py:1257]: But we don't really want them to navigate through next levels, do we?

File: app.py
Comment (hasan-sh) [reply app.py:1261]: Overall in the codebase indeed that's not the case. For all PRs i worked on, I had to find out where the variables are passed to the view and where not, that made it harder to find why certain part of the view renders or doesn't. 

I think we should comment them out to inform other devs that these may be used, see https://jinja.palletsprojects.com/en/3.0.x/templates/#variables
> If a variable or attribute does not exist, you will get back an undefined value. What you can do with that kind of value depends on the application configuration: the default behavior is to evaluate to an empty string if printed or iterated over, and to fail for every other operation.


So, if we assign an unused variable to false and we iterate over it, the rendering will fail. Whereas it wouldn't if we didn't assign it at all.

Comment (Felienne) [reply app.py:1261]: Great find, thanks! Shall we open an issue to make other renderings similar (not even for you, this is a great first issue for a newcomer too)

**Comment by Felienne** [comment]
Great work team!! 

I love how we put the next name on the button:

image

(I envisioned just "next" but this is nicer)

**Comment by Felienne** [comment]
I fixed a small issue in the tutorial, here was a similar problem in the hour-of-code:

image

So after we merge this, we really do need a cypress tests for the hour-of-code route too.

**Comment by Felienne** [comment]
The only thing still needed I think is the tracking pixel (that @hasan-sh wanted to add to this PR)

**Comment by hasan-sh** [comment]
> I fixed a small issue in the tutorial, here was a similar problem in the hour-of-code:
> 
> image
> So after we merge this, we really do need a cypress tests for the hour-of-code route too.

Great! I'll add tests today

File: hedy_content.py
Comment (Felienne) [inline hedy_content.py:267]: Perfect!

**Review by Felienne — APPROVED** [review]

**Pull request merged**
